### ISSUE TITLE
New Rule: visible-labels

### ISSUE THREAD

**Issue by jkodu** [dequelabs/axe-core#853]
# visible-labels

This rule ensures that hidden `label` elements have a visible counterpart.

**Tags:** wcag21

## Matcher

`label` element.

## Checks

Tackled in issue in auto-wcag. Refer https://github.com/auto-wcag/auto-wcag/issues/104

**Comment by WilcoFiers** [comment]
I don't think this covers the requirement. This rule is applicable to any form field that has a hidden label. Either because it gets its label from an attribute (title / aria-label) or because its actual label is hidden (for / aria-labelledby). The check than should be that at least one of the words in the label exists in an element somewhere on the page.

**Comment by dylanb** [comment]
It is not a requirement for the visible lable to be in text. I fear this will cause too many false positives.

**Comment by jkodu** [comment]
No roadmap to pick up this rule in the short-term. Can re-open an issue if necessary. Closing for now.
